AirSkyBoat/AirSkyBoat
# Besieged Train / Prepare map updates

**_I affirm:_**


## Please enter a player-facing description

Adds besieged training / prepare stages (not enabled yet)

## What does this pull request do? (Please be technical)

- Forces advanced as beastmen prepare and train
- Changes stronghold level / orders appropiately
- Adds the consecutive defeats field for beastmen
- Adds Enable Besieged toggle
- Adds DebugBesieged macro

## Steps to test these changes

Set ENABLE_BESIEGED to true, and change rates high so one can see as they adbance.

## Special Deployment Considerations

Nothing to note. Should not affect live unless ENABLE_BESIEGED is true in settings.
DB changes should not need migration as nothing is live yet.
